src/sql-parser/src/parser.rs

```diff
@@ -2257,6 +2257,25 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_default_aws_privatelink(&mut self) -> Result<WithOptionValue<Raw>, ParserError> {
+        let _ = self.consume_token(&Token::Eq);
+        let connection = self.parse_raw_name()?;
+        let mut port: Option<u16> = None;
+        if self.consume_token(&Token::LParen) {
+            self.expect_keyword(PORT)?;
+            let pos = self.peek_pos();
+            if let Ok(port_) = u16::try_from(self.parse_literal_int()?) {
+                port = Some(port_);
+            } else {
+                return parser_err!(self, pos, "Could not parse value into port");
+            };
+            self.expect_token(&Token::RParen)?;
+        };
```

Comment:
IMO, this is a bit more idiomatic Rust and lets you avoid having a mutable value that you conditionally reset

```suggestion
        let port = if self.consume_token(&Token::LParen) {
            self.expect_keyword(PORT)?;
            let pos = self.peek_pos();
            let Ok(port) = u16::try_from(self.parse_literal_int()?) {
                Some(port)
            } else {
                return parser_err!(self, pos, "Could not parse value into port");
            }
        } else {
            None
        };
```

Reply:
yeah that doesn't quite work because port is an Option<u16> and this would be return.. Some(port), None, or a Result...

Maybe I could do something similar but where port is a Result<Option<u16>,ParserError>?

Reply:
@jubrad why would it return a `Result`? The `return` returns out of the function, not the if block

Reply:
I take it all back... I'll do it (mostly this way)